src/status.rs

```diff
@@ -458,6 +464,52 @@ impl StatusBuilder {
         self
     }
 
+    pub fn spacefed(mut self, spacefed: Spacefed) -> Self {
+        self.spacefed = Some(spacefed);
+        self
+    }
+
+    pub fn add_event(mut self, event: Event) -> Self {
+        if let Some(ref mut events) = self.events {
+            events.push(event);
+        } else {
+            self.events = Some(vec![event]);
+        }
```

Comment:
I don't think that would work. The borrow checker is unfortunately not smart enough to realize that `map` and `or_else` are mutually exclusive (which is required to move `radio_show` into the vector). The whole thing is surprisingly tricky for such a supposedly simple operation :D